### PR TITLE
[serve] Add `route_prefix` flag to serve run

### PR DESCRIPTION
**Pull request by zcin**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/serve/scripts.py [hunk 1]
```diff
@@ -359,6 +359,17 @@ def deploy(config_file_name: str, address: str):
...
     ),
 )
+@click.option(
+    "--route-prefix",
+    required=False,
+    default="/",
+    type=str,
+    help=(
+        "Route prefix for the application. This should only be used "
+        "when running an application specified by import path, and "
```
Comment (edoakes) [inline python/ray/serve/scripts.py:369]:
```suggestion
        "when running an application specified by import path and "
```

File: python/ray/serve/scripts.py [hunk 2]
```diff
@@ -466,7 +478,7 @@ def run(
             client.deploy_apps(config, _blocking=False)
             cli_logger.success("Submitted deploy config successfully.")
         else:
-            serve.run(app, host=host, port=port)
+            serve.run(app, route_prefix=route_prefix, host=host, port=port)
```
Comment (edoakes) [inline python/ray/serve/scripts.py:481]: I'm now realizing we need to update this to use `serve.start` to set the `host` and `port` -- can you do that separately?

**Review by edoakes — APPROVED** [review]

**Review by GeneDer — APPROVED** [review]
:shipit:

**Comment by zcin** [comment]
@architkulkarni Could you help merge this one? 🙏

**Pull request merged**